rotkehlchen/tests/unit/decoders/test_umami.py

```diff
@@ -0,0 +1,318 @@
+from typing import TYPE_CHECKING, Final
+
+import pytest
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.chain.arbitrum_one.modules.umami.constants import (
+    CPT_UMAMI,
+    UMAMI_STAKING_CONTRACT,
+)
+from rotkehlchen.chain.evm.constants import ZERO_ADDRESS
+from rotkehlchen.chain.evm.decoding.constants import CPT_GAS
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.assets import A_ARB, A_ETH, Asset
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.evm_event import EvmEvent
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.tests.utils.ethereum import get_decoded_events_of_transaction
+from rotkehlchen.types import Location, TimestampMS, deserialize_evm_tx_hash
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.arbitrum_one.node_inquirer import ArbitrumOneInquirer
+    from rotkehlchen.types import ChecksumEvmAddress
+
+GM_USDC_WBTC_ADDRESS: Final = string_to_evm_address('0x5f851F67D24419982EcD7b7765deFD64fBb50a97')
+A_GM_USDC_WBTC: Final = Asset('eip155:42161/erc20:0x5f851F67D24419982EcD7b7765deFD64fBb50a97')
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('arbitrum_one_accounts', [['0x706A70067BE19BdadBea3600Db0626859Ff25D74']])
+@pytest.mark.parametrize('use_clean_caching_directory', [True])
+def test_umami_deposit_request(
+        arbitrum_one_inquirer: 'ArbitrumOneInquirer',
+        arbitrum_one_accounts: list['ChecksumEvmAddress'],
+) -> None:
+    tx_hash = deserialize_evm_tx_hash('0x228e4ec7b253a3609c4e28638e0281a0458f541e380d39731fb5249cccc115f5')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(evm_inquirer=arbitrum_one_inquirer, tx_hash=tx_hash)  # noqa: E501
+    user_address, timestamp, gas_amount, fee_amount, deposit_amount = arbitrum_one_accounts[0], TimestampMS(1728996957000), '0.00002373823', '0.00024', '20'  # noqa: E501
+    assert events == [
+        EvmEvent(
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ARBITRUM_ONE,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(gas_amount)),
+            location_label=user_address,
+            notes=f'Burned {gas_amount} ETH for gas',
+            tx_hash=tx_hash,
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.ARBITRUM_ONE,
+            event_type=HistoryEventType.DEPOSIT,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(fee_amount)),
+            location_label=user_address,
+            notes=f'Execution cost of {fee_amount} ETH',
```

Comment:
Since this is a fee for the deposit can you make sure it appears after the deposit? Also add the Umami keyword. "Umami execution cost of ..."